src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let valueSeparatore = separator;
+
+    if (separator === null) {
+      valueSeparatore = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      const arrayValue = this[i] === undefined
```

Comment:
Here i'd like to name the variable something like arrayItem or arrayElement, cuz value is more used in some loops or parameters for functions.